### PR TITLE
Add Travis CI and Poetry

### PR DESCRIPTION
**Pull request by lbernick**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+before_install:
+  - sudo apt-get update
+  - sudo apt-get -y install gnupg2 libgpgme-dev swig
+install:
+  - pip install --upgrade poetry
+  - poetry install
+matrix:
+  include:
+  - name: "Lint and static analysis"
+    python: 3.6
```
Comment (obi1kenobi) [inline .travis.yml]: Any particular reason not to run lint / static analysis in a newer Python version?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    script:
+      - poetry run ./scripts/copyright_line_check.sh
+      - poetry run black --check --config pyproject.toml .
+      - poetry run flake8 --config setup.cfg . d
+      - poetry run isort --check --settings-path setup.cfg -rc .
```
Comment (obi1kenobi) [inline .travis.yml:17]: Consider adding `pylint` and `bandit` as well, possibly in a later PR.

**Comment by obi1kenobi** [comment]
Are the file deletions intentional?

**Comment by lbernick** [comment]
> Are the file deletions intentional?

Yes, I deleted setup.py because it is no longer needed with poetry, and deleted one of the licenses because I accidentally added duplicate licenses with different filenames.

File: .travis.yml (hunk 1 above)
Comment (lbernick) [reply .travis.yml]: nope!

**Comment by lbernick** [comment]
One issue with poetry that I don't love: https://github.com/python-poetry/poetry/issues/2073
since we are importing and using `__version__` in `model_metadata_v3`

**Comment by obi1kenobi** [comment]
I think I'm missing a step here — I'm not sure how `__version__` relates to that issue.

**Comment by lbernick** [comment]
> I think I'm missing a step here — I'm not sure how `__version__` relates to that issue.

Sorry, should have elaborated. The package version is specified once in pyproject.toml, but the package itself uses the variable `__version__` to include in the model metadata what version of the library serialized the model. If we want to bump the package version, we need to update it in two places, or use one of the alternate suggestions on the github issue linked above.

**Comment by obi1kenobi** [comment]
I'm still confused. That Github issue seems to talk about an issue where package extras are not persisted by `poetry` across lockfiles. Here we're talking about defining our own package's version in only a single place. I'm sorry =/ I still don't see the connection between extras on dependencies of our package, and our own package's version.

**Comment by lbernick** [comment]
Oops I linked the wrong issue! https://github.com/python-poetry/poetry/issues/144

File: .pylintrc [hunk 3]
```diff
@@ -0,0 +1,643 @@
...
+    # unused-import,           # we have other tools that check for this
+    # no-member,               # false positives, especially with external dependencies
+    # no-value-for-parameter,  # click module causes false positives:
+                               # https://github.com/PyCQA/pylint/issues/259
+    # abstract-method,         # needs some cleanup first
+    # no-absolute-import,      # maybe with Python 3 support
+    # parameter_documentation, # needs some cleanup and maybe configuration
+    # unused-argument,         # needs some cleanup and per-line suppression,
+                               # buggy / unclear how to suppress only a single function
+    # unsubscriptable-object,  # TODO (Louis): Re-enable once https://github.com/PyCQA/pylint/issues/2822 is resolved
```
Comment (obi1kenobi) [inline .pylintrc]: Heh which Louis? :D Maybe don't copy this TODO

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      - poetry run pydocstyle config=setup.cfg .
+      - poetry run pylint --rcfile=.pylintrc .
+  - name: "Unit Tests"
+    python: 3.7
+    script:
+      - poetry run python -m pytest .
```
Comment (obi1kenobi) [inline .travis.yml]: Perhaps in a separate PR, I'd add tests with all python versions that this package supports. Presumably that's 3.8 as well, and possibly 3.6 too?

**Review by obi1kenobi — COMMENTED** [review]
Looks good! Other than removing the TODO, I don't think there are any blockers from merging this. Obviously there's always room for improvement but that can come in future PRs.

File: .travis.yml (hunk 2 above)
Comment (lbernick) [reply .travis.yml:17]: having some trouble getting pylint to work. I think I'll make an issue as a TODO to add pylint, bandit, and mypy.

**Comment by lbernick** [comment]
@obi1kenobi @mikeyshulman finally got builds to pass! Right now it is using black, isort, flake8, and pydocstyle for linting, and we can potentially add pylint, mypy, and bandit later on. I also have it running tests for python 3.6 and 3.7 (TF doesn't yet support 3.8 it looks like).

**Review by obi1kenobi — APPROVED** [review]
🎉

**Pull request merged**